More flake8 errors

Fix some flake8 errors seen on main:
```
:~/git/arkouda$ flake8
./tests/checkpoint_test.py:129:13: B011 Do not call assert False since python -O removes these calls. Instead callers should raise AssertionError().
./tests/checkpoint_test.py:150:13: B011 Do not call assert False since python -O removes these calls. Instead callers should raise AssertionError().
./tests/checkpoint_test.py:164:13: B011 Do not call assert False since python -O removes these calls. Instead callers should raise AssertionError().
./tests/checkpoint_test.py:174:13: B011 Do not call assert False since python -O removes these calls. Instead callers should raise AssertionError().
./tests/numpy/pdarray_creation_test.py:177:45: B023 Function definition does not bind loop variable 'shape'.
./tests/numpy/pdarray_creation_test.py:178:64: B023 Function definition does not bind loop variable 'shape'.
./tests/numpy/pdarray_creation_test.py:179:47: B023 Function definition does not bind loop variable 'shape'.
./tests/numpy/pdarray_creation_test.py:180:55: B023 Function definition does not bind loop variable 'shape'.
./tests/numpy/pdarray_creation_test.py:181:45: B023 Function definition does not bind loop variable 'shape'.
./tests/numpy/pdarray_creation_test.py:182:65: B023 Function definition does not bind loop variable 'shape'.
./tests/numpy/pdarray_creation_test.py:183:64: B023 Function definition does not bind loop variable 'shape'.
./tests/numpy/pdarray_creation_test.py:184:64: B023 Function definition does not bind loop variable 'shape'.
./tests/numpy/pdarray_creation_test.py:185:54: B023 Function definition does not bind loop variable 'local_size'.
./tests/numpy/pdarray_creation_test.py:185:66: B023 Function definition does not bind loop variable 'shape'.
./tests/optioned-server/heartbeat.py:34:13: B011 Do not call assert False since python -O removes these calls. Instead callers should raise AssertionError().
./tests/pandas/index_test.py:290:13: B015 Result of comparison is not used. This line doesn't do anything. Did you intend to prepend it with assert?
./tests/pandas/index_test.py:293:13: B015 Result of comparison is not used. This line doesn't do anything. Did you intend to prepend it with assert?
./tests/regex_test.py:77:50: B034 subn should pass `count` and `flags` as keyword arguments to avoid confusion due to unintuitive argument positions.
./tests/regex_test.py:188:54: B034 split should pass `maxsplit` and `flags` as keyword arguments to avoid confusion due to unintuitive argument positions.

s:~/git/arkouda$ flake8 --version
7.3.0 (darglint: 1.8.1, flake8-bugbear: 24.12.12, flake8-builtins: 2.5.0, flake8-docstrings: 1.7.0, flake8_import_order: 0.19.2, mccabe: 0.7.0, pycodestyle:
2.14.0, pydoclint: 0.6.6, pyflakes: 3.4.0) CPython 3.13.5 on Linux


```

--- a/tests/checkpoint_test.py
+++ b/tests/checkpoint_test.py
@@ -126,7 +126,7 @@ class TestCheckpoint:
         create_fake_cp(cp_name, num_locales=ak.get_config()["numLocales"] + 1)
         try:
             ak.load_checkpoint(cp_name)
-            assert False  # should not get here
+            raise AssertionError("Expected RuntimeError was not raised")
         except RuntimeError as err:
             assert (
                 "Attempting to load a checkpoint that was made with a different number of locales"
@@ -147,7 +147,7 @@ class TestCheckpoint:
 
         try:
             ak.load_checkpoint(cp_name)
-            assert False  # should not get here
+            raise AssertionError("Expected RuntimeError was not raised")
         except RuntimeError as err:
             assert "could not read chunk 1 metadata" in str(err)
         finally:
@@ -161,7 +161,7 @@ class TestCheckpoint:
 
         try:
             ak.load_checkpoint(cp_name)
-            assert False  # should not get here
+            raise AssertionError("Expected RuntimeError was not raised")
         except RuntimeError as err:
             assert "field 'size' not found" in str(err)
         finally:
@@ -171,7 +171,7 @@ class TestCheckpoint:
     def test_wrong_argument(self):
         try:
             ak.save_checkpoint(mode="override")
-            assert False  # should not get here
+            raise AssertionError("Expected RuntimeError was not raised")
         except ValueError as err:
             assert 'invalid checkpointing mode "override"' in str(err)
 

--- a/tests/numpy/pdarray_creation_test.py
+++ b/tests/numpy/pdarray_creation_test.py
@@ -164,6 +164,8 @@ class TestPdarrayCreation:
         assert a is c
         ak_assert_equal(a, c)
 
+    from functools import partial
+
     @pytest.mark.skip_if_max_rank_less_than(2)
     @pytest.mark.parametrize("size", pytest.prob_size)
     def test_large_array_creation_multi_dim(self, size, subtests):
@@ -173,16 +175,18 @@ class TestPdarrayCreation:
         """
         for rank in multi_dim_ranks():
             shape, local_size = _generate_test_shape(rank, size)
+
+            # Bind loop variables at definition time to satisfy Ruff B023
             test_cases = [
-                ("ak.ones", lambda: ak.ones(shape, int)),
-                ("ak.array(ak.ones)", lambda: ak.array(ak.ones(shape, int))),
-                ("ak.zeros", lambda: ak.zeros(shape)),
-                ("ak.ones (default)", lambda: ak.ones(shape)),
-                ("ak.full", lambda: ak.full(shape, 9)),  # only numeric for multi-dim
-                ("ak.zeros_like", lambda: ak.zeros_like(ak.ones(shape))),
-                ("ak.ones_like", lambda: ak.ones_like(ak.zeros(shape))),
-                ("ak.full_like", lambda: ak.full_like(ak.zeros(shape), 9)),
-                ("ak.randint", lambda: ak.randint(0, local_size, shape)),
+                ("ak.ones", lambda s=shape: ak.ones(s, int)),
+                ("ak.array(ak.ones)", lambda s=shape: ak.array(ak.ones(s, int))),
+                ("ak.zeros", lambda s=shape: ak.zeros(s)),
+                ("ak.ones (default)", lambda s=shape: ak.ones(s)),
+                ("ak.full", lambda s=shape: ak.full(s, 9)),  # only numeric for multi-dim
+                ("ak.zeros_like", lambda s=shape: ak.zeros_like(ak.ones(s))),
+                ("ak.ones_like", lambda s=shape: ak.ones_like(ak.zeros(s))),
+                ("ak.full_like", lambda s=shape: ak.full_like(ak.zeros(s), 9)),
+                ("ak.randint", lambda s=shape, n=local_size: ak.randint(0, n, s)),
             ]
 
             for name, constructor in test_cases:

--- a/tests/optioned-server/heartbeat.py
+++ b/tests/optioned-server/heartbeat.py
@@ -31,6 +31,6 @@ class TestHeartbeat:
             # 'keepalive' options in the heartbeat implementation
             # allow 4 times the timeout. Add a wiggle room of 2*timeout.
             ak.client.server_sleep(hb_timeout * 6)
-            assert False  # an exception should have been raised
+            raise AssertionError("Expected RuntimeError was not raised")
         except Exception as e:
             assert "connection to the server is closed or disconnected" in str(e)

--- a/tests/pandas/index_test.py
+++ b/tests/pandas/index_test.py
@@ -276,7 +276,7 @@ class TestIndex:
         i4_cpy = ak.Index(["a", "b", "c"], allow_list=False)
         self.assert_equal(i4 == i4_cpy, ak.array([True, True, True]))
         self.assert_equal(i4 != i4_cpy, ak.array([False, False, False]))
-        assert i3.equals(i3_cpy)
+        assert i4.equals(i4_cpy)
 
         i5 = ak.Index(["a", "x", "c"], allow_list=True)
         self.assert_equal(i3 == i5, ak.array([True, False, True]))
@@ -287,10 +287,10 @@ class TestIndex:
         assert not i5.equals(i6)
 
         with pytest.raises(ValueError):
-            i5 == i6
+            _ = i5 == i6
 
         with pytest.raises(ValueError):
-            i5 != i6
+            _ = i5 != i6
 
         with pytest.raises(TypeError):
             i.equals("string")

--- a/tests/regex_test.py
+++ b/tests/regex_test.py
@@ -74,7 +74,7 @@ class TestRegex:
         strings = ak.array(s)
         ak_sub, ak_sub_counts = strings.subn(pattern, repl, count)
         re_sub, re_sub_counts = zip(
-            *(re.subn(pattern, repl, strings[i], count) for i in range(strings.size))
+            *(re.subn(pattern, repl, strings[i], count=count) for i in range(strings.size))
         )
 
         assert ak_sub.tolist() == list(re_sub)
@@ -185,7 +185,7 @@ class TestRegex:
         maxsplit = 3
         split, split_map = strings.regex_split(pattern, maxsplit, return_segments=True)
         for i in range(strings.size):
-            re_split = re.split(pattern, strings[i], maxsplit)
+            re_split = re.split(pattern, strings[i], maxsplit=maxsplit)
             ak_split = (
                 split[split_map[i] :]
                 if i == strings.size - 1

